js/ai/src/tool.ts

```diff
@@ -187,3 +187,16 @@ export function defineTool<I extends z.ZodTypeAny, O extends z.ZodTypeAny>(
   );
   return a as ToolAction<I, O>;
 }
+
+/**
+ * Thrown when tools execution is interrupted. It's meant to be caugh by the framework, not public API.
+ */
+export class ToolInterruptError extends Error {}
+
+/**
+ * Interrupts current tool execution causing tool request to be returned in the generation response.
+ * Should only be called within a tool.
+ */
+export function interruptTool() {
```

Comment:
What do you think about adding `interrupt` as an item in the second argument of the tool instead?

```ts
ai.defineTool(..., (input, {context, interrupt}) => {
  interrupt();
});
```

Reply:
done